renderlib/RotateTool.cpp

```diff
@@ -261,6 +261,8 @@ RotateTool::draw(SceneView& scene, Gesture& gesture)
 
   gesture.graphics.addCommand(GL_LINES);
 
+  glm::vec4 discPlane(camFrame.vz, -glm::dot(camFrame.vz, axis.p));
+
```

Comment:
Could you break down what this line is doing? I'm assuming it's making a normal vector pointing towards the camera (to be the clipping plane's normal vector) but I'm not quite following how it's getting there.

Is `camFrame.vz` the vector pointing out/away from the camera, and the dot product is getting a magnitude along that vector...?

Reply:
This is standard geometry/linear algebra. The construct for a plane: a normal vector and a distance from the origin. The plane equation is Ax+By+Cz=D.    `camFrame.vz` is the view direction which I am using as the normal to the plane.  That provides the A,B,C.  And the plane is positioned at (`axis.p` dot `camFrame.vz`) units away from the eye.  That's D.  `axis.p` is the world space center of the gizmo.

When you "dot" this plane vector with a point (x,y,z,1), you get a number that is positive or negative if the point is on either side of the plane.  That's the equivalent to the value of Ax+By+Cz-D.  if it's zero, the point is on the plane.

Reply:
> ...And the plane is positioned at (`axis.p` dot `camFrame.vz`) units away from the eye...

Clarification question, should this be units away from the origin?

Reply:
Yes, my mistake- it's in world space units from the origin, not from the eye.